src/main/java/ch/njol/skript/expressions/ExprPermissions.java

```diff
@@ -18,71 +18,146 @@
 import ch.njol.util.coll.CollectionUtils;
 
-import org.bukkit.entity.Player;
-import org.bukkit.event.Event;
-import org.bukkit.permissions.PermissionAttachment;
-import org.bukkit.permissions.PermissionAttachmentInfo;
-import org.jetbrains.annotations.Nullable;
-
-import java.util.HashSet;
-import java.util.Set;
-
-@Name("All Permissions")
-@Description("Returns all permissions of the defined player(s). Note that the modifications to resulting list do not actually change permissions.")
+@Name("Permissions")
+@Description({
+	"All permissions of the provided permissible(s). A permissible is anything that can have permissions (like the player).",
+	"A couple of notes: Plugins like Skript cannot modify the default permissions such as bukkit.* permissions.",
+	"Skript also doesn't save the permissions after server restart.",
+	"You will need a permissions plugin or to save them yourself in variables."
+})
 @Examples("set {_permissions::*} to all permissions of the player")
-@Since("2.2-dev33")
-public class ExprPermissions extends SimpleExpression<String> {
-	
+@Since("2.2-dev33, INSERT VERSION (changers)")
+public class ExprPermissions extends PropertyExpression<Entity, String> {
+
 	static {
-		Skript.registerExpression(ExprPermissions.class, String.class, ExpressionType.PROPERTY, "[(all [[of] the]|the)] permissions (from|of) %players%", "[(all [[of] the]|the)] %players%'[s] permissions");
+		Skript.registerExpression(ExprPermissions.class, String.class, ExpressionType.PROPERTY,
+				"[all [[of] the]|the] permissions (from|of) %entities%",
+				"%entities%'[s] permissions"
+		);
 	}
-	
-	@SuppressWarnings("null")
-	private Expression<Player> players;
-	
-	@SuppressWarnings({"null", "unchecked"})
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		players = (Expression<Player>) exprs[0];
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		setExpr((Expression<Entity>) exprs[0]);
 		return true;
 	}
 
 	@Override
-	@Nullable
-	protected String[] get(Event e) {
-		final Set<String> permissions = new HashSet<>();
-		for (Player player : players.getArray(e))
-			for (final PermissionAttachmentInfo permission : player.getEffectivePermissions())
-				permissions.add(permission.getPermission());
-		return permissions.toArray(new String[permissions.size()]);
+	protected String[] get(Event event, Entity[] source) {
+		return getExpr().stream(event)
+				.flatMap(permissible -> permissible.getEffectivePermissions().stream())
+				.map(PermissionAttachmentInfo::getPermission)
+				.toArray(String[]::new);
 	}
-	
+
+	@Override
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+			case ADD:
+			case REMOVE:
+				return CollectionUtils.array(String[].class);
+			case DELETE:
+				return CollectionUtils.array();
+			default:
+				return null;
+		}
```

Comment:
```suggestion
		return switch (mode) {
			case ADD, REMOVE, DELETE -> CollectionUtils.array(String[].class);
			default -> null;
		};
```
We can simplify this now 🙂 
It's fine to just return the regular array for DELETE (no difference compared to an empty one)